### PR TITLE
feat(gatsby): Remove boundActionCreators

### PR DESCRIPTION
**Pull request by LekoArts**
## Description

Removes the deprecated `boundActionCreators` API.

## Related Issues

[ch23415]

### CODE REVIEW

File: packages/gatsby/src/utils/dev-ssr/render-dev-html.ts [hunk 1]
```diff
@@ -131,11 +131,11 @@ export const renderDevHTML = ({
...
     }
 
-    const { boundActionCreators } = require(`../../redux/actions`)
-    const { createServerVisitedPage } = boundActionCreators
+    const { actions } = require(`../../redux/actions`)
+    const { createServerVisitedPage } = actions
     // Record this page was requested. This will kick off adding its page
     // component to the ssr bundle (if that's not already happened)
-    createServerVisitedPage(pageObj.componentChunkName)
+    internalStore.dispatch(createServerVisitedPage(pageObj.componentChunkName))
```
Comment (wardpeet) [inline packages/gatsby/src/utils/dev-ssr/render-dev-html.ts:138]: Why aren't we using the store provided by `renderDevHTML`?

File: packages/gatsby/src/services/create-pages.ts [hunk 2]
```diff
@@ -75,7 +76,7 @@ export async function createPages({
   )
   tim.end()
 
-  boundActionCreators.apiFinished({ apiName: `createPages` })
+  internalStore.dispatch(actions.apiFinished({ apiName: `createPages` }))
```
Comment (wardpeet) [inline packages/gatsby/src/services/create-pages.ts:79]: Why aren't we using the store provided by `renderDevHTML`?

**Review by wardpeet — COMMENTED** [review]
Looking good, left 2 comments

File: packages/gatsby/src/utils/dev-ssr/render-dev-html.ts (hunk 1 above)
Comment (LekoArts) [reply packages/gatsby/src/utils/dev-ssr/render-dev-html.ts:138]: Didn't have a specific reason, just mimicked what I did in other places. Can change it to use the provided store 👍

**Review by laurieontech — DISMISSED** [review]
It looks like Ward's comments have been addressed and the code looks reasonable otherwise!

**Review by wardpeet — APPROVED** [review]

**Pull request merged**